handler/src/main/java/io/netty/handler/ssl/SslUtils.java

```diff
@@ -71,6 +74,54 @@
      */
     static final int NOT_ENCRYPTED = -2;
 
+    static final String[] DEFAULT_CIPHER_SUITES = new String[] {
+        // GCM (Galois/Counter Mode) requires JDK 8.
+        "TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384",
+        "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
+        "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
+        "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA",
+        // AES256 requires JCE unlimited strength jurisdiction policy files.
+        "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA",
+        // GCM (Galois/Counter Mode) requires JDK 8.
+        "TLS_RSA_WITH_AES_128_GCM_SHA256",
+        "TLS_RSA_WITH_AES_128_CBC_SHA",
+        // AES256 requires JCE unlimited strength jurisdiction policy files.
+        "TLS_RSA_WITH_AES_256_CBC_SHA"
+    };
+
+    /**
+     * Add elements from {@code names} into {@code enabled} if they are in {@code supported}.
+     */
+    static void addIfSupported(Set<String> supported, List<String> enabled, String... names) {
+        for (String n: names) {
+            if (supported.contains(n)) {
+                enabled.add(n);
+            }
+        }
+    }
+
+    static void useFallbackCiphersIfDefaultIsEmpty(List<String> defaultCiphers, Iterable<String> fallbackCiphers) {
+        if (defaultCiphers.isEmpty()) {
+            for (String cipher : fallbackCiphers) {
+                if (cipher.startsWith("SSL_") || cipher.contains("_RC4_")) {
+                    continue;
+                }
+                defaultCiphers.add(cipher);
+            }
+        }
+    }
+
+    static void useFallbackCiphersIfDefaultIsEmpty(List<String> defaultCiphers, String[] fallbackCiphers) {
```

Comment:
Sure I can do this ... the original motivation was to avoid the intermediate allocation/iteration/copy ... however this method is called infrequently so it shouldn't matter.